frontend/src/v5/store/tickets/comments/ticketComments.sagas.ts

```diff
@@ -0,0 +1,181 @@
+
+export function* fetchComments({
+	teamspace,
+	projectId,
+	modelId,
+	ticketId,
+	isFederation,
+}: FetchCommentsAction) {
+	try {
+		const fetchModelComments = isFederation
+			? API.TicketComments.fetchFederationComments
+			: API.TicketComments.fetchContainerComments;
+		const { comments } = yield fetchModelComments(teamspace, projectId, modelId, ticketId);
+		const richComments = sortBy(comments, 'createdAt').map(({ createdAt, updatedAt, ...comment }) => ({
+			...comment,
+			createdAt: new Date(createdAt),
+			updatedAt: new Date(updatedAt),
+		}));
+
+		yield put(TicketCommentsActions.fetchCommentsSuccess(ticketId, richComments));
+	} catch (error) {
+		yield put(DialogsActions.open('alert', {
+			currentActions: formatMessage(
+				{ id: 'comments.fetch.error', defaultMessage: 'trying to fetch the comments for {model} ticket' },
+				{ model: isFederation ? 'federation' : 'container' },
+			),
+			error,
+			details: formatMessage({
+				id: 'comments.fetch.error.details',
+				defaultMessage: 'If reloading the page doesn\'t work please contact support',
+			}),
+		}));
+	}
+}
+
+export function* createComment({
+	teamspace,
+	projectId,
+	modelId,
+	ticketId,
+	isFederation,
+	comment,
+	onSuccess,
+	onError,
+}: CreateCommentAction) {
+	try {
+		const createModelComment = isFederation
+			? API.TicketComments.createFederationComment
+			: API.TicketComments.createContainerComment;
+		const newComment = parseMessageAndImages(comment);
+
+		const { _id: newCommentId } = yield createModelComment(teamspace, projectId, modelId, ticketId, newComment);
+
+		const fetchModelComment = isFederation
+			? API.TicketComments.fetchFederationComment
+			: API.TicketComments.fetchContainerComment;
+
+		let fetchedComment = yield fetchModelComment(teamspace, projectId, modelId, ticketId, newCommentId);
+		fetchedComment = {
+			...fetchedComment,
+			createdAt: new Date(fetchedComment.createdAt),
+			updatedAt: new Date(fetchedComment.updatedAt),
+		};
+		yield put(TicketCommentsActions.upsertCommentSuccess(ticketId, fetchedComment));
+		onSuccess();
+	} catch (error) {
+		onError(error);
+		yield put(DialogsActions.open('alert', {
+			currentActions: formatMessage(
+				{ id: 'comments.create.error', defaultMessage: 'trying to create the comment for {model} ticket' },
+				{ model: isFederation ? 'federation' : 'container' },
+			),
+			error,
+			details: formatMessage({
+				id: 'comments.create.error.details',
+				defaultMessage: 'If reloading the page doesn\'t work please contact support',
+			}),
+		}));
+	}
+}
+
+export function* updateComment({
+	teamspace,
+	projectId,
+	modelId,
+	ticketId,
+	isFederation,
+	commentId,
+	comment,
+}: UpdateCommentAction) {
+	try {
+		const updateModelComment = isFederation
+			? API.TicketComments.updateFederationComment
+			: API.TicketComments.updateContainerComment;
+
+		yield updateModelComment(teamspace, projectId, modelId, ticketId, commentId, parseMessageAndImages(comment));
```

Comment:
The comment object being sent to the backend, follows this structure
```ts
{
  images: string[],
  message: string,
  ...
}
```

When the message is empty, instead of sending `{ message: '', ... }`, the `message` property should be deleted
Something similar goes for the images, if there are no images, instead of sending `{ images: [], ... }`, the `images` property should be deleted.

So before sending the new (or the updated) comment to the backend, it shall be processed as described above.

One last operation is to strip the 64BasePrefix on the images (if present).

Regarding the snippet of code that this comment refers to, you can see why it needed by doing this:
- send a comment with a message and at least one image;
- update the message so that it is empty;
- save the comment;

This will succeed, the comment will be correctly updated. But if you were to remove the `parseMessageAndImages` (line 127), the following steps will result in an error because the message property is being sent to the backend as an empty string instead of being removed from the comment object